lwgenerate/gcp/gcp.go

```diff
@@ -0,0 +1,533 @@
+
+func createGcpProvider(args *GenerateGcpTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	attrs := map[string]interface{}{}
+
+	if args.ServiceAccountCredentials != "" {
+		attrs["credentials"] = args.ServiceAccountCredentials
+	}
+
+	if args.GcpProjectId != "" {
+		attrs["project"] = args.GcpProjectId
+	}
+
+	provider, err := lwgenerate.NewProvider(
+		"google",
+		lwgenerate.HclProviderWithAttributes(attrs),
+	).ToBlock()
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+
+	return blocks, nil
+}
+
+func createLaceworkProvider(args *GenerateGcpTfConfigurationArgs) (*hclwrite.Block, error) {
+	if args.LaceworkProfile != "" {
+		return lwgenerate.NewProvider(
+			"lacework",
+			lwgenerate.HclProviderWithAttributes(map[string]interface{}{"profile": args.LaceworkProfile}),
+		).ToBlock()
+	}
+	return nil, nil
+}
+
+func createConfig(args *GenerateGcpTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	if args.Config {
+		attributes := map[string]interface{}{}
+		moduleDetails := []lwgenerate.HclModuleModifier{}
+
+		// default to using the project level module
+		configModuleName := "gcp_project_level_config"
+		if args.OrganizationIntegration {
+			// if organization integration is true, override configModuleName to use the organization level module
+			configModuleName = "gcp_organization_level_config"
+			attributes["org_integration"] = args.OrganizationIntegration
+			attributes["organization_id"] = args.GcpOrganizationId
+		}
+
+		if args.ExistingServiceAccount != nil {
+			attributes["use_existing_service_account"] = true
+			attributes["service_account_name"] = args.ExistingServiceAccount.Name
+			attributes["service_account_private_key"] = args.ExistingServiceAccount.PrivateKey
+		}
+
+		if args.ConfigIntegrationName != "" {
+			attributes["lacework_integration_name"] = args.ConfigIntegrationName
+		}
+
+		moduleDetails = append(moduleDetails,
+			lwgenerate.HclModuleWithAttributes(attributes),
+		)
+
+		moduleBlock, err := lwgenerate.NewModule(
+			configModuleName,
+			lwgenerate.GcpConfigSource,
+			append(moduleDetails, lwgenerate.HclModuleWithVersion(lwgenerate.GcpConfigVersion))...,
+		).ToBlock()
+
+		if err != nil {
+			return nil, err
+		}
+		blocks = append(blocks, moduleBlock)
+
+	}
+
+	return blocks, nil
+}
+
+func createAuditLog(args *GenerateGcpTfConfigurationArgs) (*hclwrite.Block, error) {
+	if args.AuditLog {
+		attributes := map[string]interface{}{}
+
+		moduleDetails := []lwgenerate.HclModuleModifier{}
+
+		if args.PubSubSubscriptionLabels != nil {
+			attributes["pubsub_subscription_labels"] = args.PubSubSubscriptionLabels
+		}
+
+		if args.PubSubTopicLabels != nil {
+			attributes["pubsub_topic_labels"] = args.PubSubTopicLabels
+		}
+
+		if args.ExistingLogBucketName != "" {
+			attributes["existing_bucket_name"] = args.ExistingLogBucketName
+		} else {
+			if args.LogBucketLifecycleRuleAge == nil {
+				defaultValue := -1
```

Comment:
nit: Do we need to default this? Looks like the TF module already handles this and it would kinda be not great to have to maintain this default in two places?